src/GridCal/Engine/Simulations/ShortCircuitStudies/short_circuit.py

```diff
@@ -11,29 +11,75 @@ def short_circuit_3p(bus_idx, Zbus, Vbus, Zf, baseMVA):
         Zf: Fault impedance array
 
     Returns: Voltages after the short circuit (p.u.), Short circuit power in MVA
+    Computed as V = Vpre + V increment, where Vpre is the power flow solution and
+    V increment is the fault contribution
+    The short-circuit power is V_i^2 / Z[i,i], it will tend to infinity if the
+    generator is ideal (r1 and x1 approach 0)
 
     """
     n = len(Vbus)
-    Z = Zbus[bus_idx, :][:, bus_idx]  # Z_B in documentation
 
-    # Voltage Source Contribution
-    I_kI = np.zeros(n, dtype=complex)
-    Z.flat[::len(bus_idx) + 1] += Zf[bus_idx]  # add Zf to diagonals of Z_B
-    I_kI[bus_idx] = -1 * np.linalg.solve(Z, Vbus[bus_idx])
+    Ifvec = np.zeros(n, dtype=complex)
+    Ifvec[bus_idx] = Vbus[bus_idx] / (Zbus[bus_idx, bus_idx] + Zf[bus_idx])
+    
+    Av = - Zbus @ Ifvec
+    V = Vbus + Av
 
-    # Current source contribution
-    # I_kII = -1 * Zbus.dot(I_kC / Z[elm_idx])
+    idx_buses = range(n)
+    SCC = baseMVA * np.power(np.abs(Vbus[idx_buses]), 2) / np.abs(Zbus[idx_buses, idx_buses])
 
-    # Total current contribution
-    # I_k = I_kI + I_kII
-    I_k = I_kI
+    return V, SCC
 
-    # voltage increment due to these currents
-    incV = Zbus.dot(I_k)
 
-    V = Vbus + incV
+def short_circuit_unbalance(bus_idx, Z0, Z1, Z2, Vbus, Zf, fault_type):
+    """
+    Executes the unbalanced short circuits (LG, LL, LLG types)
+
+    :param bus_idx: bus where the fault is caused
+    :param Z0: impedance matrix for the zero sequence
+    :param Z1: impedance matrix for the positive sequence
+    :param Z2: impedance matrix for the negative sequence
+    :param Vbus: prefault voltage (positive sequence)
+    :param Zf: fault impedance
+    :return: V0, V1, V2 for all buses
+    """
+
+    n = len(Vbus)
+
+    # solve the fault
+    Vpr = Vbus[bus_idx]
+    Zth0 = Z0[bus_idx, bus_idx]
+    Zth1 = Z1[bus_idx, bus_idx]
+    Zth2 = Z2[bus_idx, bus_idx]
+
+    if fault_type == 'LG':
```

Comment:
The fault types are better expressed with an Enum

Reply:
Ok, it should be fine now in the commit e22f5a8